### ISSUE TITLE
iOS VoiceControl not working when Semantics has ExcludeSemantics as child

### ISSUE THREAD

**Issue by Wasmund1** [flutter/flutter#171692]
### Steps to reproduce

1. Activate Voice Control Show Numbers 
2. Run Sample Code with Flutter 3.29.0
3. Run Sample Code with Flutter 3.32.5

### Expected results

Voice Control Show Numbers displays a label/tooltip that can be interacted with for every actionable semantic widget.

### Actual results

When a Semantics widget has an ExcludeSematics widget as child no label/tooltip is shown.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceAround,
          children: [
            SemanticWidget(label: 'Semantics'),
            SemanticWidgetWithExcludeSemantics(label: 'ExcludeSemantics'),
          ],
        ),
      ),
    );
  }
}

class SemanticWidgetWithExcludeSemantics extends StatelessWidget {
  final String label;

  @override
  Widget build(BuildContext context) {
    return Semantics(
      button: true,
      label: label,
      child: ExcludeSemantics(
        child: InkWell(
          onTap: () => print('clicked : $label'),
          child: Container(
            decoration: BoxDecoration(
              borderRadius: BorderRadius.circular(10),
              border: Border.all(width: 1, color: Colors.black),
            ),
            child: Padding(
              padding: const EdgeInsets.all(16.0),
              child: Text(label),
            ),
          ),
        ),
      ),
    );
  }

  const SemanticWidgetWithExcludeSemantics({super.key, required this.label});
}

class SemanticWidget extends StatelessWidget {
  final String label;

  @override
  Widget build(BuildContext context) {
    return Semantics(
      button: true,
      label: label,
      child: InkWell(
        onTap: () => print('clicked : $label'),
        child: Container(
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(10),
            border: Border.all(width: 1, color: Colors.black),
          ),
          child: Padding(
            padding: const EdgeInsets.all(16.0),
            child: Text(label),
          ),
        ),
      ),
    );
  }

  const SemanticWidget({super.key, required this.label});
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

| 3.29.0 | 3.32.5 |
| --- | --- |
| ![Image](https://github.com/user-attachments/assets/18b9eaaf-c4e3-4735-a5d7-fc8f6c7dbc92) | ![Image](https://github.com/user-attachments/assets/a75fc3d1-4690-4510-b339-999d7e57602c) |

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
... exclude_semantics % flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.32.5, on macOS 14.6.1 23G93 darwin-arm64, locale en-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.2)
[✓] IntelliJ IDEA Ultimate Edition (version 2024.2.4)
[✓] IntelliJ IDEA Community Edition (version 2025.1)
[✓] VS Code (version 1.101.0)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by tirth-patel-nc** [comment]
hi @Wasmund1, thanks for the report. I don't have access to an iOS device right now, but I've asked a colleague to check on theirs. in the meantime, could you please try verifying this on the latest master as well?

~also, do you know how to enable labels like these in talkback on android?~

**Comment by tirth-patel-nc** [comment]
thanks for the report. it is replicable on latest `stable 3.32.5` and `master 3.33.0-1.0.pre.849` while not in `3.29.3`. labelling it as a regression.

<img width="512" height="832" alt="Image" src="https://github.com/user-attachments/assets/1d64e9d0-8413-4d1d-a926-d344b5da0706" />

**Comment by tirth-patel-nc** [comment]
hi @Wasmund1, I am wondering why widget has to be visible for VoiceControl when it is wrapped in ExcludeSemantics 🤔 If you exclude it then you would like it to be invisible for VoiceControl?!

**Comment by Wasmund1** [comment]
Hi @tirth-patel-nc,It's not exactly wrapped in ExcludeSemantics but rather has ExcludeSemantics as a child. This is just a minimal sample to demonstrate what is happening. The Semantics widget with all it's info should be visible for assistive technology. Everything further down the widget tree should be excluded.

**Comment by chunhtai** [comment]
I think this is by design. If you wrap the ExcludeSemantics on the InkWell, the ontap action will be ignore in semantics. To voice control the button won't have any semantics action, thus inactive.
